### DISCUSSION TITLE
Requests for new attributes in VDB table

### DISCUSSION THREAD

**Discussion by dhruv-anand-aintech** [superlinked/VectorHub#163] [VDB Comparison]
This thread can server to discuss new attributes that people wish to add to the vector DB table

**Comment by dhruv-anand-aintech** [comment]
[Launched] 1. pypi package downloads (lifetime)

Easy to pull programmatically. More indicative of popularity than github stars

**Reply by dhruv-anand-aintech** [reply]
We would also require a python client package name attribute for this

**Reply by svonava** [reply]
I like this - additionally it makes sense to look at nodejs/javascript clients like:

https://www.npmjs.com/package/@pinecone-database/pinecone
https://www.npmjs.com/package/chromadb

**Reply by AruneshSingh** [reply]
This would be really informative addition! We can combine these into a single column of packages. Also wondering what the best way to implement it would be. Any thoughts?
- Should we add it as a column in the table
- Or make it in the links for the table (and shown on popup like the other links)

**Reply by svonava** [reply]
shown on popup like the other links // @AruneshSingh I suspect that ~nobody discovers this, so having click stats/hotjar would be helpful

adding it as columns of a new column group (Statistics), move github stars there and add a few other metrics would make sense because people can then sort by this

**Reply by ClaireSuperlinked** [reply]
Sounds good in general. I think we should be mindful though about adding too many new columns in a statistics view as there are already so many columns that it's quite easy for people to be overwhelmed already (which is reflected in the feedback). Let's get some hotjar data and use people's requests to evaluate adding columns (like Dhruv's which kicks off this thread) :)

**Reply by AruneshSingh** [reply]
Launched this in the latest update.

**Comment by dhruv-anand-aintech** [comment]
[Launched] 2. Docker pulls

Again, an objective measure of popularity. Can be pulled programmatically

**Reply by AruneshSingh** [reply]
Launched this in the latest update.

**Reply by svonava** [reply]
@dhruv-anand-aintech how do we separate the done from the open in this list?

**Reply by AruneshSingh** [reply]
I've gone ahead and added a [Launched] at the start of the comments to indicate.

**Comment by dhruv-anand-aintech** [comment]
3. SDK Language support

Another list attribute to show which languages have maintained SDKs. Perhaps more useful than development_languages

**Reply by svonava** [reply]
Like this one too - useful for users to figure out what they can work with easily, given their environment!

**Comment by dhruv-anand-aintech** [comment]
Support for Diversity reranking (by metadata/content)
https://python.langchain.com/docs/integrations/vectorstores/astradb#mmr-maximal-marginal-relevance-search

**Reply by svonava** [reply]
Any idea how this actually works? The docs don't have anything besides the method reference. Also I wonder how many vendors offer something like this.

**Comment by dhruv-anand-aintech** [comment]
Free managed cloud instance:
1. Disk/RAM size
2. Number of days of idling before instance is hibernated
3. Number of days of idling before instance is deleted

**Reply by svonava** [reply]
This could be interesting, but a nightmare to update. What would be our strategy to keep this up to date, if the vendors are not so quick with the updates?

**Comment by dhruv-anand-aintech** [comment]
VectorIO implementation:
1. Import 
2. Export

**Comment by dhruv-anand-aintech** [comment]
DB Admin Panel: Yes/No

Features:
- Visualization (like Qdrant has)
- Connect to non-cloud instance (like weaviate cloud console has)
- add more

**Comment by dhruv-anand-aintech** [comment]
Scroll API: ability to iterate through all data points in the index sequentially

**Comment by dhruv-anand-aintech** [comment]
Serverless offering

**Comment by dhruv-anand-aintech** [comment]
Client protocols supported: checkboxes: HTTP, gRPC, GraphQL

**Comment by dhruv-anand-aintech** [comment]
Index Types: https://www.linkedin.com/feed/update/urn:li:activity:7158028192027418628/

![Screenshot 2024-02-02 at 1 02 16 PM](https://github.com/superlinked/VectorHub/assets/105786647/09fc7981-d80e-4895-86f7-7d6de12e218b)

**Reply by svonava** [reply]
this one is a must have!

**Reply by svonava** [reply]
@dhruv-anand-aintech let's launch this one?

**Reply by svonava** [reply]
perhaps we could somehow use this column to integrate what library the vendor is using - for example ClickHouse & DuckDB now use usearch as far as I know (https://github.com/unum-cloud/usearch)

**Reply by dhruv-anand-aintech** [reply]
working on it now at https://github.com/superlinked/VectorHub/pull/386

**Comment by dhruv-anand-aintech** [comment]
USP column: feature that is only checked in this row (or very few other rows)

**Comment by dhruv-anand-aintech** [comment]
Community link/invite: discord/slack

**Comment by dhruv-anand-aintech** [comment]
[Launched] int8 and binary embeddings storage support
https://txt.cohere.com/int8-binary-embeddings/

**Reply by svonava** [reply]
Now launched right? We should mark the suggestions here as done to keep track.

**Comment by dhruv-anand-aintech** [comment]
CDC (change data capture) connector

**Comment by Kain-90** [comment]
[Launched] Multi-user attribute support, I believe this will be very helpful for all businesses that need to provide a private knowledge base for each users

**Reply by svonava** [reply]
This already exists - look for the "Multi-tenant" column:
<img width="183" alt="image" src="https://github.com/superlinked/VectorHub/assets/1493333/99edbac6-6895-4901-91a8-a94d60582c52">

**Comment by dhruv-anand-aintech** [comment]
Easy paginated queries (eg: offset:5, limit:10)

**Comment by amaan784** [comment]
Can you add Databricks Vector Database or Mosaic Vector Search to the list?

**Reply by dhruv-anand-aintech** [reply]
Hi Amaan, this is an open source repo, so the DB vendors can submit PRs for adding the data.

Also, this discussion thread is for new columns/features in the table.
